lib/hanami/model/sql/consoles/mysql.rb

```diff
@@ -0,0 +1,65 @@
+require 'shellwords'
+
+module Hanami
+  module Model
+    module Sql
+      module Consoles
+        # MySQL adapter
+        #
+        # @since x.x.x
+        # @api private
+        class Mysql
+          # @since x.x.x
+          # @api private
+          def initialize(uri)
+            @uri = uri
+          end
+
+          # @since x.x.x
+          # @api private
+          def connection_string
+            str = 'mysql'
+            str << host
+            str << database
+            str << port if port
+            str << username if username
+            str << password if password
+            str
```

Comment:
@cllns it's not the same (note the statement modifiers/conditionals).

But I still agree with your solution:-)

`String#<<` will not work with `frozen_string_literal`

So it should be

```ruby
parts = []
parts << "something"
parts << "else" if condition
parts.join # is the connection string concatenated with empty string?
```